### ISSUE TITLE
A problem when reproduced HyperTransformer code

### ISSUE THREAD

**Issue by fyoung26** [wgcban/HyperTransformer#8]
Hello Chaminda,

Thank you for the code and work on HyperTransformer. When I tried to reproduce the score on the botswana4 dataset, the metrics I got are far from expectation:

pretrain:
{"loss": 0.07473030593246222, "cc": 0.9290400743484497, "sam": 3.0020320415496826, "rmse": 0.021659649908542633, "ergas": 0.6483104825019836, "psnr": 27.892120361328125}

train：
{"loss": 0.08555552270263433, "cc": 0.8993576765060425, "sam": 3.35520076751709, "rmse": 0.02654602937400341, "ergas": 0.7366535663604736, "psnr": 26.585630416870117}

I also used the trained model you provided, and I got:
{"loss": 0.05360260047018528, "cc": 0.9539724588394165, "sam": 2.2932522296905518, "rmse": 0.01636636257171631, "ergas": 1.8692368268966675, "psnr": 30.393962860107422}

Both results are far from expectation. 

Then,I checked the github issue of HyperTransformer, then I changed "max_value": 8000 to "max_value": 9816(I got this value from the proccess code of matlab),the pretrain metrics got improvement:

{"loss": 0.03306201007217169, "cc": 0.964657187461853, "sam": 1.863145351409912, "rmse": 0.01357241254299879, "ergas": 0.3927982747554779, "psnr": 32.014068603515625}

But still far away from expectation. 

Do you know how to solve this problem?

**Comment by fyoung26** [comment]
I tried to contact you via email(wbandar1@jhu.edu), but dont get reply, I really need your help, thank you.T T

**Comment by wgcban** [comment]
@fyoung26 I am not exactly sure what is the reason for lower scores on the Botswana dataset. Let me look into it.

**Comment by fyoung26** [comment]
@wgcban Thank you sincerely, I'm looking for your reply💖

**Comment by maozan** [comment]
Hi @fyoung26 . May I ask, did you solve your puzzle?

**Comment by fyoung26** [comment]
> Hi @fyoung26 . May I ask, did you solve your puzzle?

Hi @maozan , I didn't solve it yet, do you have any idea about?

**Comment by EchoPhD** [comment]
> @fyoung26我不确定博茨瓦纳数据集得分较低的原因是什么。让我调查一下。

Hello Sir, 
Thank you for the code and work on HyperTransformer. I am very confused about the data set processed by matlab. Could you please send me a copy of the processed data set? Email: xiaozhengwang95@gmail.com. Thank you very much.

**Comment by EchoPhD** [comment]
@wgcban 
Hello Sir,
Thank you for the code and work on HyperTransformer. I am very confused about the data set processed by matlab. Could you please send me a copy of the processed data set? Email: xiaozhengwang95@gmail.com . Thank you very much.

**Comment by wgcban** [comment]
@HaiMaoShiTang Hi, I am not sure about I still have the dataset saved in my machine since I am not continuing right now, but I will check and respond to you tomorrow. Thanks.

**Comment by EchoPhD** [comment]
Hello, thank you very much for your reply. It will help me a lot. Thank you very much.







——————————————————————————————
Xiaozheng Wang    Machine Learning/ Remote Sensing Image Fusion
School of Control Science and Engineering,  Tiangong University.
Tianjin,  China.
E-mail: ***@***.***

发件人: Chaminda Bandara
发送时间: 2023年3月12日 21:53
收件人: wgcban/HyperTransformer
抄送: HaiMaoShiTang; Mention
主题: Re: [wgcban/HyperTransformer] A problem when reproducedHyperTransformer code (Issue #8)

@HaiMaoShiTang Hi, I am not sure about I still have the dataset saved in my machine since I am not continuing right now, but I will check and respond to you tomorrow. Thanks.

**Comment by hachreak** [comment]
Hi @wgcban ,
it looks like there are many issues with the code and also with the pretrained.

E.g. dataset padova:

I downloaded your [pretrained](https://www.dropbox.com/sh/9zg0wrbq6fzx1wa/AACH3mnRlqkVFmo6BF4wcDdaa?dl=0) of N_modules 8 for pavia and I tried to compute the PSNR, running the test function.

The test function returns a PSNR of **41.60820770263672** but in the paper you wrote **43.80**!

How can I replicate your results? 
I'm doing something wrong? 
Could you please share the your datasets?
Thanks a lot for your help!

**Comment by ChenJFU** [comment]
> 您好，非常感谢您的回复。这对我有很大帮助。谢谢。 —————————————————————————————— 王晓政 机器学习/遥感影像融合 天宫大学控制科学与工程学院. 中国天津。 电子邮件： ***@***.*** 发件人： 查敏达·班达拉 发送时间: 2023年3月12日 21:53 收件人： wgcban/HyperTransformer 抄送： 海茂石堂;提到 主题： 回复： [wgcban/HyperTransformer] 复制HyperTransformer 代码时出现问题（问题 #8） @HaiMaoShiTang 嗨，我不确定我是否仍然将数据集保存在我的机器中，因为我现在不继续，但我明天会检查并回复您。谢谢。 — 直接回复此电子邮件，在 GitHub 上查看或取消订阅。 您收到此消息是因为您被提及。消息 ID：***@***.***>



> 您好先生，感谢您的代码和HyperTransformer的工作。我对 matlab 处理的数据集感到非常困惑。您能否将已处理数据集的副本发送给我？电子邮件： [xiaozhengwang95@gmail.com](mailto:xiaozhengwang95@gmail.com) .谢谢。



> Hello Sir,
> Thank you for the code and work on HyperTransformer. I am very confused about the data set processed by matlab. Could you please send me a copy of the processed data set? Email: [xiaozhengwang95@gmail.com](mailto:xiaozhengwang95@gmail.com). Thank you very much.
Hello Sir,
Thank you for the code and work on HyperTransformer. I'm also confused about the preprocessed data set. Could you please send me a copy of the processed data set? My Email: chenjiafu113@gmail.com. Thank you very much.

**Comment by ChenJFU** [comment]
> @HaiMaoShiTang Hi, I am not sure about I still have the dataset saved in my machine since I am not continuing right now, but I will check and respond to you tomorrow. Thanks.

Hello Sir,
Thank you for the code and work on HyperTransformer. I'm also confused about the preprocessed data set. Could you please send me a copy of the processed data set? My Email: [chenjiafu113@gmail.com](mailto:chenjiafu113@gmail.com). Thank you very much.
